test/cluster/config_test.go

```diff
@@ -0,0 +1,419 @@
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+	leaderServer := c.Servers[leaderIndex]
+
+	//try to delete the only admin in the cluster
+	configEnv, err := leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig := configEnv.GetResponse().GetConfig()
+	newConfig.Admins = []*types.Admin{}
+	version := configEnv.GetResponse().GetMetadata().GetVersion()
+
+	_, _, err = leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.EqualError(t, err, "failed to submit transaction, server returned: status: 400 Bad Request, message: Invalid config tx, reason: admin config is empty. There must be at least single admin in the cluster")
+
+	//// add admin2 to admins list
+	err = c.CreateAdditionalUserCryptoMaterials("admin2")
+	require.NoError(t, err)
+	certKeyPair, err := getUserCert("admin2", c)
+	require.NoError(t, err)
+	require.NotNil(t, certKeyPair)
+
+	admin2 := []*types.Admin{
+		{
+			Id:          "admin2",
+			Certificate: certKeyPair.Certificate[0],
+		},
+	}
+	//get current cluster config
+	configEnv, err = leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig = configEnv.GetResponse().GetConfig()
+
+	newConfig.Admins = append(newConfig.Admins, admin2...)
+	version = configEnv.GetResponse().GetMetadata().GetVersion()
+
+	//add admin2
+	txID, rcpt, err := leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
+
+	//make sure admin2 in the admins list
+	configEnv, err = leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	admins := configEnv.GetResponse().GetConfig().GetAdmins()
+	require.Equal(t, 2, len(admins))
+	require.Equal(t, admin2[0], admins[1])
+
+	//remove original admin
+	newConfig = configEnv.GetResponse().GetConfig()
+	newConfig.Admins = admin2
+	version = configEnv.GetResponse().GetMetadata().GetVersion()
+
+	txID, rcpt, err = leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
+
+	admin2CertPath, admin2KeyPath := c.GetUserCertKeyPath("admin2")
```

Comment:
before switching to the 2nd admin, make sure the `admin` you removed can no longer Query of Submit a config TX.